eng/Update-PkgVersion.ps1

```diff
@@ -65,78 +63,39 @@ function Parse-Version($VersionString)
         Write-Error "Missing or Empty Version property ${VersionString}"
         exit 1
     }
+    $semVer = [SemVer]::new($VersionString)
 
-    if ($VersionString -Match $VERSION_REGEX)
+    if ($semVer.IsPrerelease -eq $true -and $semVer.PrereleaseLabel -ne 'preview')
```

Comment:
May be worth adding these common version checks to the SemVer class.

Reply:
Adding as part of https://github.com/Azure/azure-sdk-tools/pull/472